controllers/applicationvolumereplication_controller.go

```diff
@@ -120,106 +126,263 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+	if r.isSubsriptionPausedForDR(subscription.GetLabels()) {
+		if err := r.processPausedSubscription(avr, subscription); err != nil {
+			r.Log.Error(err, fmt.Sprintf("failed to process paused Subscription %s", subscription.Name))
+		}
+		// Stop processing this subscription. We'll wait for the next Reconciler iteration
+		r.Log.Info("Skipping paused subscription", "name", subscription.Name)
 
-	avr.Status = ramendrv1alpha1.ApplicationVolumeReplicationStatus{
-		Decisions: placementDecisions,
+		return nil, requeue
 	}
-	if err := r.Client.Status().Update(ctx, avr); err != nil {
-		return errorswrapper.Wrap(err, "failed to update AVR status")
+
+	// Skip this subscription if a manifestwork already exist
+	if found, err := r.findVRGManifestWork(avr, subscription); err != nil {
+		return nil, requeue
+	} else if found {
+		r.Log.Info("Skipping existing mainifestwork for subscription", "name", subscription.Name)
+
+		return nil, !requeue
+	}
+
+	// This subscription is ready for manifest (VRG) creation
+	r.Log.Info("Subscription is unpaused", "Name", subscription.Name)
+
+	placementDecision, err := r.processUnpausedSubscription(subscription)
+	if err != nil {
+		return nil, requeue
+	}
+
+	return &placementDecision, !requeue
+}
+
+func (r *ApplicationVolumeReplicationReconciler) isSubsriptionPausedForDR(labels map[string]string) bool {
+	return labels != nil &&
+		labels[RamenDRLabelName] != "" &&
+		strings.EqualFold(labels[RamenDRLabelName], "protected") &&
+		labels[subv1.LabelSubscriptionPause] != "" &&
+		strings.EqualFold(labels[subv1.LabelSubscriptionPause], "true")
+}
+
+// processPausedSubscription selects the target cluster from the subscription or
+// from the user selected cluster and restores all PVs (belonging to the subscription)
+// to the target cluster and then it unpauses the subscription
+func (r *ApplicationVolumeReplicationReconciler) processPausedSubscription(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) error {
+	r.Log.Info("Processing paused subscription", "name", subscription.Name)
+	// find target cluster (which can be the failover cluster)
+	homeClusterName := r.findNextHomeCluster(avr, subscription)
+
+	if homeClusterName == "" {
+		return errorswrapper.New("failed to find new home cluster")
+	}
+
+	err := r.deleteExistingManfiestWork(avr, subscription)
+	if err != nil {
+		return err
+	}
+
+	err = r.restoreBackedupPVs(subscription, homeClusterName)
+	if err != nil {
+		return err
+	}
+
+	return r.unpauseSubscription(subscription)
+}
+
+func (r *ApplicationVolumeReplicationReconciler) findVRGManifestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) (bool, error) {
+	const notFound = false
+
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		mw := &ocmworkv1.ManifestWork{}
+		vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, "vrg")
```

Comment:
Shouldn't we check if home/peer clusters are as per the placement decisions of the subscription as well? IOW, look to see if Subscription intends to be placed in the homeCluster as of now.

(future) we possibly should shift to looking at MW in MC namespace on the hub to determine what the ASoW for the VRG/Subscription is (as suggested by you Benamar in a past discussion). Helps avoid Status based decisions in the future.